### PR TITLE
iphone: keep the example usable for ios

### PR DESCRIPTION
**Pull request by noondie**
hello 
here is my first pull request for wxWidgets. 
I tried to update the iPhone branch and it works with catalina in the simulator
Unfortunately i also pushed my test file (minimal.cpp) into this branch. I guess you can easily ignore that file.

### CODE REVIEW

File: include/wx/private/spinctrl.h [hunk 1]
```diff
@@ -10,6 +10,11 @@
 #ifndef _WX_PRIVATE_SPINCTRL_H_
 #define _WX_PRIVATE_SPINCTRL_H_
 
+#ifdef __WXOSX_IPHONE__
+#include "wx/gdicmn.h"
+#include "wx/control.h"
+#endif
+
```
Comment (vadz) [inline include/wx/private/spinctrl.h]: I think this file is not supposed to be used at all under iOS because there is no `wxSpinCtrl` there AFAICS. So I don't really understand why is this necessary. Could you please explain what was the error which required this change?

File: src/osx/carbon/dcscreen.cpp [hunk 2]
```diff
@@ -26,12 +26,14 @@ wxIMPLEMENT_ABSTRACT_CLASS(wxScreenDCImpl, wxWindowDCImpl);
 wxScreenDCImpl::wxScreenDCImpl( wxDC *owner ) :
    wxWindowDCImpl( owner )
 {
+#ifndef __WXOSX_IPHONE__
```
Comment (vadz) [inline src/osx/carbon/dcscreen.cpp]:
```suggestion
#if !wxOSX_USE_IPHONE
```
for consistency with the other checks in this file.

File: src/osx/carbon/metafile.cpp [hunk 3]
```diff
@@ -92,12 +92,13 @@ wxMetafileRefData::wxMetafileRefData(CFDataRef data) :
 wxMetafileRefData::wxMetafileRefData( const wxString& filename )
 {
     Init();
-
+#ifndef __WXOSX_IPHONE__
```
Comment (vadz) [inline src/osx/carbon/metafile.cpp]: It looks like it should be possible to avoid this, as the code is there since 2008 and must have compiled under iOS until f83577df451. Again, knowing what the exact problem was would be helpful here.

File: src/osx/core/sound.cpp [hunk 4]
```diff
@@ -134,6 +134,7 @@ bool wxSound::Create(const wxString& fileName, bool isResource)
 {
     wxCHECK_MSG( !isResource, false, "not implemented" );
 
+#ifndef __WXOSX_IPHONE__
     wxCFRef<CFURLRef> url(wxOSXCreateURLFromFileSystemPath(fileName));
```
Comment (vadz) [inline src/osx/core/sound.cpp]: Again, I think it should be possible to provide `wxOSXCreateURLFromFileSystemPath()` for iOS instead.

**Review by vadz — CHANGES_REQUESTED** [review]
Thanks for your PR but unfortunately some changes look questionable to me.

I'd like to fix iOS build, of course, but ideally I'd like to do it without disabling the existing functionality as in metafile and sound case.

File: src/osx/carbon/dcscreen.cpp (hunk 2 above)
Comment (noondie) [reply src/osx/carbon/dcscreen.cpp]: agreed , fine. i will change this in the upcoming commit

File: src/osx/carbon/metafile.cpp (hunk 3 above)
Comment (noondie) [reply src/osx/carbon/metafile.cpp]: problem here is 
../src/osx/carbon/metafile.cpp:99:31: error: use of undeclared identifier 'wxOSXCreateURLFromFileSystemPath'
        wxCFRef<CFURLRef> url(wxOSXCreateURLFromFileSystemPath(filename));

wxOSXCreateURLFromFileSystemPath is only declared in include/wx/osx/core/private.h:WXDLLIMPEXP_BASE CFURLRef wxOSXCreateURLFromFileSystemPath( const wxString& path);
but in src/common/filefn.cpp the rule is the following. 
#if defined(__WXMAC__) && !defined(__WXOSX_IPHONE__)

any hint to solve this ??

File: src/osx/core/sound.cpp (hunk 4 above)
Comment (noondie) [reply src/osx/core/sound.cpp]: like in the metafile.cpp example, we need to know, why there  is a distinction between MACOS and OSX.

File: include/wx/private/spinctrl.h (hunk 1 above)
Comment (noondie) [reply include/wx/private/spinctrl.h]: this was also not my preferred solution . 
the problem was 
In file included from ../src/common/spinctrlcmn.cpp:28:
../include/wx/private/spinctrl.h:32:8: error: unknown type name 'wxSize'
extern wxSize GetBestSize(const wxControl* spin, int minVal, int maxVal, int base);
       ^
../include/wx/private/spinctrl.h:32:33: error: unknown type name 'wxControl'
extern wxSize GetBestSize(const wxControl* spin, int minVal, int maxVal, int base);

it seems, that these both header files are not included in the build path.

File: src/osx/carbon/metafile.cpp (hunk 3 above)
Comment (vadz) [reply src/osx/carbon/metafile.cpp]: Sorry for not being clear: my idea is that `wxOSXCreateURLFromFileSystemPath()` should be made available under iOS too, as it's just a wrapper for `CFURLCreateWithFileSystemPath()` which seems to be available there. So I'd rename it to something (@csomor what prefix do you use for functions available under both macOS and iOS?) and move it outside of `!wxOSX_USE_IPHONE` check.

File: include/wx/private/spinctrl.h (hunk 1 above)
Comment (vadz) [reply include/wx/private/spinctrl.h]: This one is simple to fix so I just did this blindly in 6c061b6bf91e8cd825c28e0af9875431a3e6d7b0

File: src/osx/carbon/metafile.cpp (hunk 3 above)
Comment (csomor) [reply src/osx/carbon/metafile.cpp:95]: unfortunately there is no common name prefix in use. I've used `wxOSX` due to the common ancestry for all platforms.

Comment (csomor) [reply src/osx/carbon/metafile.cpp:95]: I'll make this available in core, thus for all platforms,

Comment (csomor) [reply src/osx/carbon/metafile.cpp:95]: in eaec76ee97435a36a295d7f79139e747cbd6d45e

**Comment by vadz** [comment]
@noondie Could you please check the latest master, Stefan's change should have fixed at least some of the problems you've encountered, so could you please try rebuilding again to see what is left to fix?

BTW, it would be really great if somebody (you or someone else...) could add iOS build to our `.travis.yml` -- there is no reason we couldn't build it under Travis macOS, is there? I'm not sure which [Xcode version](https://docs.travis-ci.com/user/reference/osx/#xcode-version) would it be best to use for this build though, any ideas?

**Comment by noondie** [comment]
hello
thanks for your help. a quick check gives an idea, that it works. recently i have checked the ios build. I will do this also for the macos build in the next days.
@vadz i never have done a yaml file but will try it on eastern. one more reason to stay at home ;=) 

Maybe someone can give me a hint how to proceed with this task. Shall i close this or will this be done by anybody else.

**Comment by vadz** [comment]
This really depends on whether there is anything remaining to be done or not... If you say that iOS builds on master now, this PR should be closed. If you can, please make another one adding Travis build later. TIA!

File: src/osx/cocoa/utils_base.mm [hunk 5]
```diff
@@ -205,6 +205,8 @@ wxString wxGetOsDescription()
 }
 #endif // wxUSE_DATETIME
 
+#ifndef __WXOSX_IPHONE__
+// this is not allowed in IOS
```
Comment (csomor) [inline src/osx/cocoa/utils_base.mm:209]: I've committed that part already

**Comment by noondie** [comment]
everything seems to be done
